app/Controllers/FileController.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controllers;
+
+class FileController
+{
+    private string $file_name;
+    private string $file_path;
+    private $raw_data;
+    private array $formatted_data;
+
+    public function __construct(string $file_name)
+    {
+
+        $this->file_name = $file_name;
+
+        $this->file_path = $this->createFilePath();
+    }
```

Comment:
Not a blocker, if you use constructor property promotion feature, try to be consistent across files. In this case, move the declarion on line 9 to be a promoted property (private string $file_name)

Reply:
Done